### PR TITLE
Remove occurrences of `?` as a glob

### LINKED ISSUE #644: Check uses of "?" for fish 3 compatibility
Howdy,

it appears OMF uses the "?" glob in a few places. Fish 3 deprecates that one and allows users to disable it via a "feature flag" (like the `^` stderr redirection).

Going through the ones I've found via `git grep '\?' | grep -v .md:`:

These should presumably just use `*` instead:

```fish
init.fish:  set -l key_bindings $OMF_CONFIG/key_binding?.fish \
init.fish:                      {$OMF_CONFIG,$OMF_PATH}/themes/$theme/key_binding?.fish
pkg/omf/functions/themes/omf.theme.set.fish:    and test (count {$OMF_CONFIG,$OMF_PATH}/key_binding?.fish) -gt 0
```

These read the part after the "=" into a variable immediately afterwards, and the variable defaults to empty, so the `?` can presumably simply be dropped:

```fish
pkg/omf/functions/index/omf.index.query.fish:      case '--name=?*'
pkg/omf/functions/index/omf.index.query.fish:      case '--text=?*'
```

I'd probably just drop the `-?` and `/?` way of saying "--help":

```fish
bin/install:      case --help -h '/?'
pkg/omf/functions/omf.fish:      case '-h' '--help' '-\?' '/\?'
```

If you must offer it, matching a literal `?` here is a bit more involved than I'd like it, but since it doesn't happen often.... This is the code we use:

```fish
    # A literal "?" for use in `case`.
    set -l q '\\?'
    if status test-feature qmark-noglob
        set q '?'
    end

    # ...
    case "$q"
```

This is just an error for an unknown option, so it's probably also okay to trigger on just a "-":

```
pkg/omf/functions/omf.fish:      case '-?*'
```

(Alternatively, catch `case -` before, and then use `case '-*'`)

### PR DESCRIPTION
**Pull request by scorphus**
Since Fish Shell 3.0 [1], `?` as a glob is deprecated. Fixes #644

  1. https://github.com/fish-shell/fish-shell/releases/tag/3.0.0

### CODE REVIEW

File: bin/install [hunk 1]
```diff
@@ -29,7 +29,7 @@ function main
   # Parse command args
   while set -q argv[1]
     switch "$argv[1]"
-      case --help -h '/?'
+      case --help -h
```
Comment (bobthecow) [inline bin/install]: these aren't a glob, but seems reasonable to remove all the `-?` and `/?` args anyway?

File: init.fish [hunk 2]
```diff
@@ -33,9 +33,9 @@ function fish_user_key_bindings
     and read -l theme < $OMF_CONFIG/theme
     or set -l theme default
   # Prepare packages key bindings paths
-  set -l key_bindings $OMF_CONFIG/key_binding?.fish \
+  set -l key_bindings $OMF_CONFIG/key_binding*.fish \
                       {$OMF_CONFIG,$OMF_PATH}/pkg/*/key_bindings.fish \
-                      {$OMF_CONFIG,$OMF_PATH}/themes/$theme/key_binding?.fish
+                      {$OMF_CONFIG,$OMF_PATH}/themes/$theme/key_binding*.fish
```
Comment (bobthecow) [inline init.fish]: Switching from `?` to `*` in these leads to a much greater chance of false positives (maybe someone has a `key_bindings_old.fish` file or something) so I don't love it. Maybe moving the `*` into the path somewhere is a better approach?

```
$OMF_CONFIG*/key_bindings.fish \
  {$OMF_CONFIG,$OMF_PATH}/pkg/*/key_bindings.fish \
  {$OMF_CONFIG,$OMF_PATH}/themes*/$theme/key_bindings.fish
```

File: bin/install (hunk 1 above)
Comment (scorphus) [reply bin/install]: Yup, not a glob. I think it's awkward and not very common. That's my humble opinion, though. And I guess you agree on that?

File: init.fish (hunk 2 above)
Comment (scorphus) [reply init.fish]: I didn't think of that, good catch! I agree that's a better approach and the result is the same. Thanks!

File: bin/install (hunk 1 above)
Comment (sagebind) [reply bin/install]: I vote just remove it. It is the Windows way of asking for help, but people running OMF on Windows these days are just using WSL or cygwin anyway.

File: init.fish [hunk 3]
```diff
@@ -33,9 +33,9 @@ function fish_user_key_bindings
     and read -l theme < $OMF_CONFIG/theme
     or set -l theme default
   # Prepare packages key bindings paths
-  set -l key_bindings $OMF_CONFIG/key_binding?.fish \
+  set -l key_bindings $OMF_CONFIG/key_binding*.fish \
```
Comment (bobthecow) [inline init.fish]: should we move the `*` on this or the `off.theme.set` at the bottom?

Comment (scorphus) [reply init.fish]: I'm looking into that.

Also, it's `key_bindings.fish`, with a trailing `s`, at least for the plugins and themes that have it:

* [plugin-bang-bang](/oh-my-fish/plugin-bang-bang)
* [plugin-expand](/oh-my-fish/plugin-expand)
* [plugin-sudope](/oh-my-fish/plugin-sudope)
* [theme-budspencer](/oh-my-fish/theme-budspencer)
* [theme-dangerous](/oh-my-fish/theme-dangerous)

Comment (scorphus) [reply init.fish]: Having `$OMF_CONFIG*/key_bindings.fish` leads to problem in case of `~/.config/omf_old`, for example.

Comment (scorphus) [reply init.fish]: Maybe there's no way to make it part of `$key_bindings` like the others. What if we solve it with the following?

```fish
test -e $OMF_CONFIG/key_bindings.fish; and source $OMF_CONFIG/key_bindings.fish
```

**Comment by scorphus** [comment]
I'll run a few more tests tomorrow, I'm too tired now.

**Comment by scorphus** [comment]
Tests completed. Versions tested:
Fish 2.3.1 ✅
Fish 2.7.1 ✅
Fish 3.0.0 ✅
Fish 3.0.1 ✅

🐟 🐚 FTW

**Comment by scorphus** [comment]
@bobthecow, thanks for your reviews. When you have the time, please, I'd really like to read your inputs regarding the last two changes (they will be autosquashed). Thanks!

**Pull request merged**